java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -746,14 +747,61 @@ public Description matchBinary(BinaryTree tree, VisitorState state) {
       fixBuilder.replace(tree, remainingSubTree.toString());
       decrementAllSymbolUsages(deletedSubTree, state, fixBuilder);
       builder.addFix(fixBuilder.build());
-
       endPos = state.getEndPosition(tree);
       return builder.build();
     }
 
     return Description.NO_MATCH;
   }
 
+  @Override
+  public Description matchParenthesized(ParenthesizedTree tree, VisitorState state) {
+    if (overLaps(tree, state)) {
+      return Description.NO_MATCH;
+    }
+
+    ExpressionTree expression = tree.getExpression();
```

Comment:
The `evalExpr` method (see [here](https://github.com/uber/piranha/blob/eb73d95a36b4a8cb92e8db4e1a3cb7bf3dc27b1a/java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java#L498)) performs the evaluation of the expression. Also, there is some overlap of code in `matchBinary`  and in this method. Can you please consider refactoring this change to see: 
a) whether it could be handled within `evalExpr` itself? 
b) if not, have a common helper for `matchBinary` and `matchParenthesized` and update `matchBinary` too? 

Also, to check the generalization of the solution, can you add more tests with parenthesis? examples: 
1. `if(a && (b && (isTreated(...) || c)`
2. `if((a && !isControl(...)) && c)`
3. `if(a && !(b && isTreated(...))`
4. `if((a && isTreated(...)) || (b && isTreated(...))`
5. ...
...